2-api/server.js

```diff
@@ -0,0 +1,275 @@
+const express = require("express");
+const app = express();
+
+app.use(express.json());
+
+const PORT = process.env.PORT || 5000;
+
+const { Pool } = require("pg");
+
+const pool = new Pool({
+  user: "postgres",
+  host: "localhost",
+  database: "cyf_ecommerce",
+  password: "",
+  port: 5432,
+});
+
+app.get("/", (req, res) => {
+  res.json("Welcome to the cyf_ecommerce server");
+});
+
+app.get("/customers", (req, res) => {
+  pool.query("SELECT * FROM customers", (err, result) => {
+    if (!err) {
+      res.status(200).json(result.rows);
+    } else {
+      res.status(400);
+      console.error(err);
+    }
+  });
+});
+
+app.get("/customers/:customerId", (req, res) => {
+  const customerId = req.params.customerId;
+  pool.query(
+    "SELECT * FROM customers WHERE id=$1",
+    [customerId],
+    (err, result) => {
+      if (!err) {
+        res.status(200).json(result.rows);
+      } else {
+        res.status(400);
+        console.error(err);
+      }
+    }
+  );
+});
+
+app.get("/customers/:customerId/orders", (req, res) => {
+  const customerId = req.params.customerId;
+  pool
+    .query(
+      "SELECT orders.order_reference, orders.order_date, products.product_name, product_availability.unit_price, supplier.supplier_name, order_items.quantity FROM order_items INNER JOIN orders on order_id = orders.id INNER JOIN customers ON customer_id = customers.id INNER JOIN product_availability ON prod_id=product_id INNER JOIN suppliers ON supp_id = suppliers.id INNER JOIN products ON prod_id = products.id WHERE customers.id=$1",
+      [customerId]
```

Comment:
You need to join order items with product availability on both product id and supplier id. Also there's a typo -  supplier.supplier_name should be suppliers.supplier_name